File: src/main/java/top/theonly/spark/jav/core/partition/MyPartitioner.java
package top.theonly.spark.jav.core.partition;

import org.apache.spark.Partitioner;

import java.io.Serializable;

/**
 * 自定义Spark分区其器
 * @author theonly
 */
public class MyPartitioner extends Partitioner implements Serializable {
    @Override
    public int numPartitions() {
        return 6;
    }

    @Override
    public int getPartition(Object key) {
        Integer k = Integer.valueOf(key.toString());
        return k % numPartitions();
    }
}

File: src/main/scala/top/theonly/spark/sca/core/partition/MyPartitioner.scala
package top.theonly.spark.sca.core.partition

import org.apache.spark.Partitioner

/**
 * 自定义Spark分区其器
 */
object MyPartitioner extends Partitioner {
  override def numPartitions: Int = 6

  override def getPartition(key: Any): Int = {
    val k: Int = key.toString.toInt
    k % Integer.MAX_VALUE
  }
}
